docs/adr/0002-shopping-lists-lineitem-and-textlineitem-update-actions.md

```diff
@@ -0,0 +1,336 @@
+          }
+        },
+        "fields": {
+          "textField": "text-2"
+        }
+      },
+     "addedAt": "2020-11-04T09:38:35.571Z"
+    }
+  ]
+}
+</pre>
+        </td>
+    </tr>
+    <tr>
+        <th colspan="2">Analysis</th>
+    </tr>
+    <tr>
+        <td colspan="2">
+            <p>Draft has line items with <b>SKU-1</b> and <b>SKU-2</b> also in target project line item with
+                <b>SKU-2</b> exists, so <b>SKU-1</b> is a new line item. </p>
+            <p> So we need to create an <a href="https://docs.commercetools.com/api/projects/shoppingLists#add-lineitem">AddLineItem</a> action
+                and a <a href="https://docs.commercetools.com/api/projects/shoppingLists#change-lineitems-order">Change LineItems Order</a> 
+                of the line items <b>SKU-1</b> and <b>SKU-2</b>, because when we add line item with <b>SKU-1</b>
+                the order will be <b>SKU-2</b> and <b>SKU-1</b>.</p>
+            <p>The <b>challenge</b> in here is, those actions can not be added in one request because we don't know the
+                line item id of the new line item
+                with <b>SKU-1</b>, so we need to find another way to create a new line item with the right order.</p>
+        </td>
+    </tr>
+    <tr>
+        <th colspan="2">Proposed solution</th>
+    </tr>
+    <tr>
+        <td colspan="2">
+            <p>
+                Normally, for a difference, we might do a set intersection and then calculate action for differences, 
+                but that does not make sense because we are not aware of the order from the draft. 
+                So in this case the one request could be created but we might need to remove line item with <b>SKU-2</b>
+                and line items in the draft with the given order with the line items <b>SKU-1</b> and <b>SKU-2</b>.
+            </p>
+            <pre lang="json">
+{
+  "version": 1,
+  "actions": [
+    {
+      "action": "removeLineItem",
+      "lineItemId": "24de3821-e27d-4ddb-bd0b-ecc99365285f"
+    },
+    {
+      "action": "addLineItem",
+      "sku": "SKU-1",
+      "quantity": 1,
+      "addedAt": "2020-11-04T09:38:35.571Z",
+      "custom": {
+        "type": {
+          "key": "custom-type-for-shoppinglists"
+        },
+        "fields": {
+          "textField": "text-1"
+        }
+      }
+    },
+    {
+      "action": "addLineItem",
+      "sku": "SKU-2",
+      "quantity": 2,
+      "addedAt": "2020-11-04T09:40:12.341Z",
+      "custom": {
+        "type": {
+          "key": "custom-type-for-shoppinglists"
+        },
+        "fields": {
+          "textField": "text-2"
+        }
+      }
+    }
+  ]
+}
+</pre>
+        </td>
+    </tr>
+</table>
+
+### Do we need to remove all line items when the order changes ?
+
+<table>
+    <tr>
+        <th>LineItemDrafts</th>
+        <th>LineItems</th>
+    </tr>
+    <tr>
+        <td><pre lang="json">
+{
+  "lineItems": [
+    {
+      "sku": "SKU-1",
+      "quantity": 1,
```

Comment:
Is quantity supposed to update? I see the quantity of existing lineitem 'sku-1' is different.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/614/commits/b8f09477fe196d172980c0bcaa05a1292eb0d7c2